Library/Homebrew/software_spec.rb

```diff
@@ -405,17 +411,27 @@ def skip_relocation?
     cellar == :any_skip_relocation
   end
 
+  sig { params(tag: Symbol).returns(T::Boolean) }
   def tag?(tag)
     checksum_for(tag) ? true : false
   end
 
-  # Checksum methods in the DSL's bottle block optionally take
+  # Checksum methods in the DSL's bottle block take
   # a Hash, which indicates the platform the checksum applies on.
+  # Example bottle block syntax:
+  # bottle do
+  #  sha256 "69489ae397e4645..." => :big_sur, :cellar => :any_skip_relocation
+  #  sha256 "449de5ea35d0e94..." => :catalina, :cellar => :any
+  # end
+  # Example args:
+  # {"69489ae397e4645..."=> :big_sur, :cellar=>:any_skip_relocation}
   def sha256(val)
-    digest, tag = val.shift
-    collector[tag] = Checksum.new(digest)
+    digest, tag = val.to_a.find { |k, _| k.is_a?(String) && k.match?(/^[a-f0-9]{64}$/i) }
```

Comment:
```suggestion
    sha256_regex = /^[a-f0-9]{64}$/i
    digest, tag = val.find do |key, value|
      key.is_a?(String) && value.is_a?(Symbol) && key.match?(sha256_regex)
    end
```

but `sha256_regex` could even be made a constant somewhere.